nixos/modules/services/networking/wireguard.nix

```diff
@@ -134,14 +164,17 @@ let
 
   };
 
-  generateConf = name: values: pkgs.writeText "wireguard-${name}.conf" ''
+  generateConf = name: values: ''
     [Interface]
-    PrivateKey = ${values.privateKey}
+    PrivateKey = ${if values.privateKeyFile != null then "$(cat ${values.privateKeyFile})" else values.privateKey}
     ${optionalString (values.listenPort != null)   "ListenPort = ${toString values.listenPort}"}
 
-    ${concatStringsSep "\n\n" (map (peer: ''
+    ${concatStringsSep "\n\n" (map (peer:
+    assert (peer.presharedKeyFile != null) != (peer.presharedKey != null);
+    ''
     [Peer]
     PublicKey = ${peer.publicKey}
+    ${optionalString (peer.presharedKeyFile != null) "PresharedKey = $(cat ${peer.presharedKeyFile})"}
```

Comment:
For PrivateKey, you do `PrivateKey = ${if values.privateKeyFile != null then "$(cat ${values.privateKeyFile})" else values.privateKey}`, which is nice since it's clear and one line. However, for PresharedKey, you do two lines.

Can you decide on one strategy so that this is consistent?